### ISSUE TITLE
[Bug]: Attempting to send NFT throws error

### ISSUE THREAD

**Issue by darkwing** [MetaMask/metamask-extension#17692]
### Describe the bug

I attempted to send an NFT on mainnet for a demo and saw the following JS error:

```
Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: 0x168ee.8
```

<img width="525" alt="NFTSend" src="https://user-images.githubusercontent.com/46655/217880411-5fd3b11c-4c48-45cb-8990-20ccdb30e505.png">

This is on current `develop` branch.

### Steps to reproduce

1.  Click on an NFT
2. Click the "send" button
3.  Attempt to send
4. See the transaction immediately fail
5. Hover over the "failed" status to see the JS error

### Version

10.25

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by brad-decker** [comment]
This may be fixed by #17651 -- Numeric.ts has a round method that was previously checking for the existence of 'numberOfDecimals' before performing the operation but 'numberOfDecimals' can be `0` which evaluates to `false` and was not rounding appropriately. @segun added a proper check and test cases for the round method. This may resolve this issue but I do not have NFTs in a wallet on mainnet that I can test with.

**Comment by darkwing** [comment]
Fixed by the PR cited.  Thank you!
